instrumentation/activities/context.go

```diff
@@ -0,0 +1,19 @@
+package activities
+
+import "context"
+
+const contextKey = "horsehead.activities.reporter"
```

Comment:
I would use an unexported type defined in this package for the contextKey. It prevents collisions unlike strings.

```go
type contextKey int
const (
    reporterKey contextKey = iota
)

context.WithValue(ctx, reporterKey, reporter)
ctx.Value(reporterKey)
```